### PR TITLE
Fixes spectral normalization chainerx conversion

### PR DESCRIPTION
**Pull request by emcastillo**
When doing spectral normalization, if using chainerx 
the link gamma parameter is not being converted back to chainerx after the fallback due to the lack of svd.

This PR proposes to use @toslunar solution in #7455 to avoid this issue.

This PR enables #7520 spectral_normalization tests to complete without errors.

### CODE REVIEW

**Comment by niboshi** [comment]
Jenkins, test this please

File: chainer/link_hooks/spectral_normalization.py [hunk 1]
```diff
@@ -247,10 +247,12 @@ def _prepare_parameters(self, link, input_variable=None):
                 else:
                     with chainer.using_device(device):
                         _, s, _ = xp.linalg.svd(array)
+                s = fallback._to_chx(s)
             else:
                 _, s, _ = link.xp.linalg.svd(weight_matrix)
+                s0 = chainer.utils.force_array(s[0])
```
Comment (niboshi) [inline chainer/link_hooks/spectral_normalization.py]: You should decrease the indent level by one.

```
2019-07-04 17:38:53.134692 STDOUT 2298] | >               link.gamma = variable.Parameter(s0)
-- | --
2019-07-04 17:38:53.134705 STDOUT 2298] | E               UnboundLocalError: Parameterized test failed.
```

Comment (emcastillo) [reply chainer/link_hooks/spectral_normalization.py]: yeah, I just fixed it :(

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**